examples/advanced/federated-statistics/hierarchical_stats/README.md

```diff
@@ -0,0 +1,245 @@
+
+```bash
+    cp /tmp/nvflare/hierarchical_stats/server/simulate_job/statistics/hierarchical_stats.json $NVFLARE_HOME/examples/advanced/federated-statistics/hierarchical_stats/demo/hierarchical_stats.json
+
+    cd $NVFLARE_HOME/examples/advanced/federated-statistics/hierarchical_stats/demo
+
+    jupyter notebook visualization.ipynb
+```
+You should be able to get the visualization similar to the followings
+
+Expandable global and level wise stats, clicking on each will shows statistics for that particular level.
+
+<img src="demo/expandable_stats.png" alt="Expandable Hierarchical stats" width="400">
+
+Expanding `Global` will show visualization of global statistics similar to the following
+
+<img src="demo/global_tats.png" alt="Global stats" width="400">
+
+Following visualization shows example state level global statistics for `State 2`
+
+<img src="demo/states_hierarchy.png" alt="States" width="400">
+
+and
+
+<img src="demo/state_level_global_stats.png" alt="State 2 stats" width="400">
+
+And the visualization of local stats at last hierarchical level should looks similar to the following
+
+<img src="demo/local_hierarchical_stats.png" alt="Local stats" width="400">
+
+
+For histograms section of `visualization.ipynb`, you should be able to get the visualization similar to the followings
+
+Global Histogram for `Percentage`
+
+<img src="demo/global_histogram.png" alt="Global Histogram" width="400">
+
+State level histogram for `Percentage`
+
+<img src="demo/state_level_histogram.png" alt="State level histogram" width="400">
+
+And local histogram for `Percentage`
+
+<img src="demo/local_histogram.png" alt="Local histogram" width="400">
+
+## 5. Configuration and Code
+
+The latest Flare has support for hierarchical statistics(server side hierarchical statistics controller and client side executor) for the federated
+statistics computing in a hierarchical way, we will only need to provide the followings
+* config_fed_server.json (server side controller configuration)
+* config_client_server.json (client side executor configuration)
+* hierarchy_config.json (Hierarchy specification file providing details about all the clients and their hierarchy.)
+* local statistics calculator
+
+### 5.1 server side configuration
+
+```
+  "workflows": [
+    {
+      "id": "fed_stats_controller",
+      "path": "nvflare.app_common.workflows.hierarchical_statistics_controller.HierarchicalStatisticsController",
+      "args": {
+        "statistic_configs": {
+          "count": {},
+          "sum": {},
+          "max": {},
+          "min": {},
+          "mean": {},
+          "var": {},
+          "stddev": {},
+          "histogram": {
+            "*": { "bins": 2, "range": [0, 1] },
+            "Percentage": {"bins": 4, "range": [0.0, 100.00] }
+          }
+        },
+        "writer_id": "stats_writer",
+        "enable_pre_run_task": false,
+        "precision": 4,
+        "hierarchy_config": "hierarchy_config.json"
```

Comment:
could you add some explanation on any requirements/structure of this "hierarchy_config"?
Does it have to be a JSON file? Is there any key that must exist etc?

Reply:
Done, thanks. Sorry I missed this important part earlier.